test/exchange_adapters/kucoin.test.ts

```diff
@@ -0,0 +1,246 @@
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+        {
+          symbol: 'CELO-USDT',
+          symbolName: 'CELO-USDT',
+          buy: undefined,
+          sell: undefined,
+          changeRate: '0.0907',
+          changePrice: '0.0629',
+          high: '0.8281',
+          low: '0.6654',
+          vol: undefined,
+          volValue: undefined,
+          last: undefined,
+          averagePrice: '0.75703415',
+          takerFeeRate: '0.001',
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+      ],
+    },
+  }
+
+  describe('parseTicker', () => {
+    it('handles a response that matches the documentation', () => {
+      const ticker = kucoinAdapter.parseTicker(validMockTickerJson)
+      expect(ticker).toEqual({
+        source: Exchange.KUCOIN,
+        symbol: kucoinAdapter.standardPairSymbol,
+        ask: new BigNumber(0.7563),
+        baseVolume: new BigNumber(1598294.854),
+        bid: new BigNumber(0.7555),
+        lastPrice: new BigNumber(0.7561),
+        quoteVolume: new BigNumber(1213224.94637127),
+        timestamp: 1674501725001,
+      })
+    })
+
+    it('throws an error when a json field mapped to a required ticker field is missing or empty', () => {
+      expect(() => {
+        kucoinAdapter.parseTicker(inValidMockTickerJson)
+      }).toThrowError('timestamp, bid, ask, lastPrice, baseVolume not defined')
+    })
+  })
+
+  describe('fetchTrades', () => {
+    it('returns empty array', async () => {
+      expect(await kucoinAdapter.fetchTrades()).toEqual([])
+    })
+  })
+
+  const validMockStatusJson = {
+    code: '200000',
+    data: [
+      {
+        symbol: 'LOKI-BTC',
+        name: 'OXEN-BTC',
+        baseCurrency: 'OXEN',
+        quoteCurrency: 'BTC',
+        feeCurrency: 'BTC',
+        market: 'BTC',
+        baseMinSize: '0.1',
+        quoteMinSize: '0.00001',
+        baseMaxSize: '10000000000',
+        quoteMaxSize: '99999999',
+        baseIncrement: '0.0001',
+        quoteIncrement: '0.000000001',
+        priceIncrement: '0.000000001',
+        priceLimitRate: '0.1',
+        minFunds: '0.000001',
+        isMarginEnabled: false,
+        enableTrading: true,
+      },
+      {
+        symbol: 'CELO-USDT',
+        name: 'CELO-USDT',
+        baseCurrency: 'CELO',
+        quoteCurrency: 'USDT',
+        feeCurrency: 'USDT',
+        market: 'USDS',
+        baseMinSize: '0.1',
+        quoteMinSize: '0.1',
+        baseMaxSize: '10000000000',
+        quoteMaxSize: '99999999',
+        baseIncrement: '0.0001',
+        quoteIncrement: '0.0001',
+        priceIncrement: '0.0001',
+        priceLimitRate: '0.1',
+        minFunds: '0.1',
+        isMarginEnabled: true,
+        enableTrading: true,
+      },
+    ],
+  }
+  const inValidMockStatusJson = {
+    code: '200000',
+    data: [
+      {
+        symbol: 'LOKI-BTC',
```

Comment:
why a second pair here for order book status check?

Reply:
To ensure that the book status function parses the correct pair.
The API response returns information for all pairs

Reply:
alright 👍